File: backend/app/schemas/posicao.py
from __future__ import annotations

from decimal import Decimal
from uuid import UUID

from pydantic import BaseModel, ConfigDict


class PosicaoRead(BaseModel):
    id: UUID
    carteira_id: UUID
    conta_id: UUID
    ativo_id: UUID

    quantidade: Decimal
    preco_medio: Decimal
    custo_total: Decimal

    model_config = ConfigDict(from_attributes=True)

File: frontend/src/types/posicao.ts
// FILE: frontend/src/types/posicao.ts

export interface Posicao {
  id: string; // UUID
  carteira_id: string; // UUID
  conta_id: string; // UUID
  ativo_id: string; // UUID
  quantidade: number;
  preco_medio: number;
  custo_total: number;
}